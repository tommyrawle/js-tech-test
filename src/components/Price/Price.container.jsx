import { connect } from 'react-redux';
import Price from './Price.presentational.jsx';

const mapStateToProps = state => ({
  oddsFormat: state.oddsFormat
});

export default connect(mapStateToProps)(Price);
