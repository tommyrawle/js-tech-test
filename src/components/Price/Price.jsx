import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { PriceWrapper } from './Price.styles.jsx';

export const Price = ({ price, oddsFormat }) => {
  return oddsFormat === 'fraction' ? (
    <PriceWrapper>{`${price.num}/${price.den}`}</PriceWrapper>
  ) : (
    <PriceWrapper>{parseFloat(price.decimal).toFixed(2)}</PriceWrapper>
  );
};

Price.propTypes = {
  price: PropTypes.object,
  oddsFormat: PropTypes.string
};

const mapStateToProps = state => ({
  oddsFormat: state.oddsFormat
});

export default connect(mapStateToProps)(Price);
