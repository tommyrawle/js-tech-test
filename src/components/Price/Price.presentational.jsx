import React from 'react';
import PropTypes from 'prop-types';
import { PriceWrapper } from './Price.styles.jsx';

const Price = ({ price, oddsFormat }) => {
  return oddsFormat === 'fraction' ? (
    <PriceWrapper>{`${price.num}/${price.den}`}</PriceWrapper>
  ) : (
    <PriceWrapper>{parseFloat(price.decimal).toFixed(2)}</PriceWrapper>
  );
};

Price.propTypes = {
  price: PropTypes.object,
  oddsFormat: PropTypes.string
};

export default Price;
