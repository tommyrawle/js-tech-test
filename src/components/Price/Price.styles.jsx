import styled from 'styled-components';

export const PriceWrapper = styled.div`
  color: #d92231;
`;
