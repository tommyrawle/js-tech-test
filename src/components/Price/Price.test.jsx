import React from 'react';
import { shallow } from 'enzyme';
import Price from './Price.presentational.jsx';

it('expect to render Price component', () => {
  const mockPrice = { price: { den: 7, num: 1, decimal: 1.1428571428571428 } };
  const mockOddsFormat = 'fraction';
  expect(shallow(<Price price={mockPrice} oddsFormat={mockOddsFormat} />)).toMatchSnapshot();
});

it('expect to render Price component', () => {
  const mockPrice = { price: { den: 7, num: 1, decimal: 1.1428571428571428 } };
  const mockOddsFormat = 'decimal';
  expect(shallow(<Price price={mockPrice} oddsFormat={mockOddsFormat} />)).toMatchSnapshot();
});
