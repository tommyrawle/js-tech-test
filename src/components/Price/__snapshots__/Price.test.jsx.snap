// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Price component 1`] = `
<styled.div>
  undefined/undefined
</styled.div>
`;

exports[`expect to render Price component 2`] = `
<styled.div>
  NaN
</styled.div>
`;
